gateway/webhook/webhook.go

```diff
@@ -122,6 +125,53 @@ func (webhook *HandleT) RequestHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	var postFrom url.Values
+	var multipartForm *multipart.Form
+
+	if r.Method == "GET" {
+		return
+	}
+	contentType := r.Header.Get("Content-Type")
+	if contentType == "application/x-www-form-urlencoded" {
+		if err := r.ParseForm(); err != nil {
+			http.Error(w, "Could not parse form", 400)
+			return
+		}
+		postFrom = r.PostForm
+	} else if strings.Contains(contentType, "multipart/form-data") {
+		if err := r.ParseMultipartForm(32 << 20); err != nil {
+			http.Error(w, "Could not parse multipartForm", 400)
+			return
+		}
+		multipartForm = r.MultipartForm
+	}
+
+	var jsonByte []byte
+	var err error
+
+	if r.MultipartForm != nil {
+		jsonByte, err = json.Marshal(multipartForm)
+		if err != nil {
+			http.Error(w, "Could not marshal form data", 400)
+			return
+		}
+	} else if len(postFrom) != 0 {
+		jsonByte, err = json.Marshal(postFrom)
+		if err != nil {
+			http.Error(w, "Could not marshal form data", 400)
+			return
+		}
+	}
+
+	if len(postFrom) != 0 || multipartForm != nil {
+		r, err = http.NewRequest("POST", r.URL.String(), bytes.NewBuffer(jsonByte))
```

Comment:
we might lose headers like x-forwarded-for that contain the request IP address etc when we create a new request. Should we edit the existing request or copy over the headers to new one